csrc/ir/base_nodes.h

```diff
@@ -524,6 +524,10 @@ class NVF_API Expr : public Statement {
       const ExpressionEvaluator& ee,
       const std::vector<PolymorphicValue>& inputs) const;
 
+  virtual std::vector<PolymorphicValue> evaluate(
```

Comment:
There should be a code comment here about why this version of `evaluate` (and thus the added complexity) is needed and how it's different from the version above.

Reply:
Added. Should be more clear now.

Reply:
Can you fix the format of the comment? It shows some unnecessary indentation.